src/test/java/io/jenkins/plugins/analysis/core/model/SourcePrinterTest.java

```diff
@@ -117,6 +117,24 @@ void shouldFilterTagsInMessageAndDescription() {
         assertThat(document.getElementsByClass("analysis-detail").html())
                 .isEqualToIgnoringWhitespace("Hello <b>Description</b>");
     }
+    
+    @Test
+    void shouldAddBreakOnNewLine() {
+        IssueBuilder builder = new IssueBuilder();
+        Issue issue = builder.setLineStart(7).setMessage("Hello <b>MessageLine1\nLine2\nLine3</b>").build();
+
+        SourcePrinter printer = new SourcePrinter();
+
+        Document document = Jsoup.parse(printer.render(asStream("format-java.txt"), issue,
+                NO_DESCRIPTION, ICON_URL));
+
+        assertThatCodeIsEqualToSourceText(document);
+
+        assertThat(document.getElementsByClass("analysis-warning-title").html())
+                .isEqualTo("<pre>Hello <b>MessageLine1</b></pre>\n<pre>Line2</pre>\n<pre>Line3</pre>");
```

Comment:
I wonder how the closing `</b>` can be before the Line3?

Reply:
So I think this has to do with the sanitizer passing in the html. It will read <b> in the first line, applying <b> to the whole line but because there is no closing tag, it doesn’t apply it to all three lines. When the closing </b> is read, it gets ignored. I can this being a problem. The best solution I can think of is perhaps having a stack for tags and applying adding that stack to each new line if not closed. I’ll write this up and add it to my branch.